src/main/java/com/mucommander/main/AutoProcessor.java

```diff
@@ -95,18 +97,15 @@ public class AutoProcessor
      * @param configMap Map of configuration properties.
      * @param context The system bundle context.
     **/
-    public static void process(Map<String, String> configMap, BundleContext context)
+    public static void process(@Nonnull Map<String, String> configMap, BundleContext context)
     {
-        ExecutorService executor = null;
-        try {
-            executor = Executors.newFixedThreadPool(8);
-            var configMapLocal = (configMap == null) ? new HashMap<String, String>() : configMap;
-            processAutoDeploy(executor, configMapLocal, context);
-            processAutoProperties(executor, configMapLocal, context);
-        } finally {
-            if (executor != null) {
-                executor.shutdown();
-            }
+        var executor = Executors.newFixedThreadPool(8);
+        try (AutoCloseable closable = executor::shutdown){
```

Comment:
or simply:
`try (var executor = Executors.newFixedThreadPool(8)) {' 

However, Executors became AutoCloseable in java 19, right?

Reply:
right